src/software/new_geom/rectangle.h

```diff
@@ -106,5 +106,23 @@ class Rectangle : public ConvexPolygon
      */
     bool expand(double amount);
 
+    /**
+     * Returns the Rectangle expanded in the direction of (x,0)
+     * The Rectangle is split along a vertical line through the center and points on the
+     * side that (x,0) points in are translated by (x,0)
+     *
+     * @return a Rectangle expanded in the direction of (x,0)
+     */
+    Rectangle expandXDirection(double x) const;
+
+    /**
+     * Returns the Rectangle expanded in the direction of (0,y)
+     * The Rectangle is split along a horizontal line through the center and points on the
+     * side that (0,y) points in are translated by (0,y)
+     *
+     * @return a Rectangle expanded in the direction of (0,y)
+     */
+    Rectangle expandYDirection(double y) const;
```

Comment:
I don't want to deal with shrinking anything so the interface cannot allow things to shrink. I rather not resort to throwing std::invalid_argument if possible

Reply:
What if I change `expand(double amount)` to `legacyExpand(double amount)` and make it deprecated?

Reply:
I'm not sure if I agree that a rectangle with a negative expansion should shrink. I think a .shrink() more explicitly defines that behaviour.

There are cases when the sign of the expansion (and there for the direction of expansion of the rectangle) matters.

@garethellis0 @MathewMacDougall

Reply:
I think the problem is that `expand(double amount)` had shrinking behaviour and having functions with the same name in the same class have shrinking and non-shrinking behaviours is confusing. I hope renaming things to  `legacyAdditiveSizeChange`, `expandXMinOrXMax`, and `expandYMinOrYMax` makes things clearer

Reply:
IMO `expandXMinOrXMax` is still a bit of a confusing name. Upon reading this again, I agree that I would definitely not assume a negative number changes which points are expanded in which direction. Because we can't have `unsigned doubles`, it might be impossible to completely avoid someone accidentally shrinking something.

Reply:
> it might be impossible to completely avoid someone accidentally shrinking something.

In general, it's impossible to avoid someone misusing _any_ of our functions if they misread or don't bother reading the function comments. For example, sometimes if I don't read the function comments, I misuse std:: functions or other geom functions, but that's on me, not the writer of the function